http/handlers.go

```diff
@@ -268,44 +292,39 @@ func writeJSON(w http.ResponseWriter, status int, data interface{}) error {
 	}
 
 	w.WriteHeader(status)
-	w.Header().Set("Content-Type", "application/json")
 	_, err = w.Write(body)
 	return err
 }
 
-func jsonError(w http.ResponseWriter, status int, err error, msg string) error {
+func jsonError(logger log.Logger, w http.ResponseWriter, status int, err error, msg string) error {
 	logger.Error(err, msg)
 	resp := map[string]interface{}{
 		"error": map[string]interface{}{
 			"error":   err.Error(),
 			"comment": msg,
 		},
 	}
-	return writeJSON(w, status, resp)
-}
-
-func handleSubscriptions(w http.ResponseWriter, r *http.Request) {
-	var getid string
-	if strings.HasPrefix(r.URL.Path, "/subscriptions/") {
-		getid = strings.TrimPrefix(r.URL.Path, "/subscriptions/")
-	}
-	hegelServer.SubLock().RLock()
-	defer hegelServer.SubLock().RUnlock()
-	var err error
-	if getid == "" {
-		err = writeJSON(w, http.StatusOK, hegelServer.Subscriptions)
-	} else if sub, ok := hegelServer.Subscriptions()[getid]; ok {
-		err = writeJSON(w, http.StatusOK, sub)
-	} else {
-		err = jsonError(w, http.StatusNotFound, errors.Errorf("%s not found", getid), "item not found")
-	}
-	if err != nil {
-		logger.Error(err)
-	}
+	return writeJSON(logger, w, status, resp)
 }
 
-func buildSubscriberHandlers() {
-	handler := otelhttp.WithRouteTag("/subscriptions", http.HandlerFunc(handleSubscriptions))
-	http.Handle("/subscriptions", handler)
-	http.Handle("/subscriptions/", handler)
+// todo(chrisdoherty4) Re-write this. It violates several laws and is dangerously fragile.
+// Also, does this work for `/subscriptions`? The writeJSON call injects a function that isn't
```

Comment:
We do seem to patch paths without trailing slashes in some places. Not sure if its in all places.

I'll circle round and remove this because if its not working, its not being used. Are you OK with that given its been fine until today. I intend on circling round on testing soon.